src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -601,17 +602,34 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                             {
                                 assert(emitter::optGetSveInsOpt(op2Size) == INS_OPTS_SCALABLE_D);
                                 sopt = INS_SCALABLE_OPTS_WIDE;
-                                break;
                             }
-
-                            FALLTHROUGH;
+                            break;
                         }
 
+                        case NI_Sve_ShiftRightArithmeticForDivide:
+                            hasShift = true;
+                            break;
+
                         default:
-                            sopt = INS_SCALABLE_OPTS_NONE;
                             break;
                     }
 
+                    auto emitInsHelper = [&](regNumber reg1, regNumber reg2, regNumber reg3) {
+                        if (hasShift)
+                        {
+                            HWIntrinsicImmOpHelper helper(this, intrinEmbMask.op2, op2->AsHWIntrinsic());
+                            for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
+                            {
+                                GetEmitter()->emitInsSve_R_R_I(insEmbMask, emitSize, reg1, reg2, helper.ImmValue(), opt,
```

Comment:
For table generated variant, can you verify using `JitStress` what code looks like for special cases below (where we use `movprfx`, etc.)?

Reply:
This is what I'm getting for the reflection scenario, without any JitStress:
```
G_M55611_IG01:  ;; offset=0x0000
            stp     fp, lr, [sp, #-0x10]!
            mov     fp, sp
                                                ;; size=8 bbWeight=1 PerfScore 1.50
G_M55611_IG02:  ;; offset=0x0008
            uxtb    w0, w0
            sub     w1, w0, #1
            cmp     w1, #8
            bhs     G_M55611_IG12
            ptrue   p0.b
            movprfx z0.b, p0/z, z0.b
            adr     x1, [G_M55611_IG03]
            add     x1, x1, x0,  LSL #3
            sub     x1, x1, #8
            br      x1
                                                ;; size=40 bbWeight=1 PerfScore 9.50
G_M55611_IG03:  ;; offset=0x0030
            asrd    z0.b, p0/m, z0.b, #1
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG04:  ;; offset=0x0038
            asrd    z0.b, p0/m, z0.b, #2
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG05:  ;; offset=0x0040
            asrd    z0.b, p0/m, z0.b, #3
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG06:  ;; offset=0x0048
            asrd    z0.b, p0/m, z0.b, #4
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG07:  ;; offset=0x0050
            asrd    z0.b, p0/m, z0.b, #5
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG08:  ;; offset=0x0058
            asrd    z0.b, p0/m, z0.b, #6
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG09:  ;; offset=0x0060
            asrd    z0.b, p0/m, z0.b, #7
            b       G_M55611_IG11
                                                ;; size=8 bbWeight=1 PerfScore 4.00
G_M55611_IG10:  ;; offset=0x0068
            asrd    z0.b, p0/m, z0.b, #8
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG11:  ;; offset=0x006C
            ldp     fp, lr, [sp], #0x10
            ret     lr
                                                ;; size=8 bbWeight=1 PerfScore 2.00
G_M55611_IG12:  ;; offset=0x0074
            bl      CORINFO_HELP_THROW_ARGUMENTOUTOFRANGEEXCEPTION
            brk_windows #0
                                                ;; size=8 bbWeight=0 PerfScore 0.00
```

Reply:
And with `JitStress=2`:
```
G_M55611_IG01:  ;; offset=0x0000
            stp     fp, lr, [sp, #-0x20]!
            mov     fp, sp
            movz    x1, #0x5678
            movk    x1, #0x1234 LSL #16
            movk    x1, #0xDEF0 LSL #32
            movk    x1, #0x9ABC LSL #48
            str     x1, [fp, #0x18]     // [V04 GsCookie]
                                                ;; size=28 bbWeight=1 PerfScore 4.50
G_M55611_IG02:  ;; offset=0x001C
            uxtb    w0, w0
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG03:  ;; offset=0x0020
            sub     w1, w0, #1
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG04:  ;; offset=0x0024
            cmp     w1, #8
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG05:  ;; offset=0x0028
            bhs     G_M55611_IG36
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG06:  ;; offset=0x002C
            ptrue   p0.b
                                                ;; size=4 bbWeight=1 PerfScore 2.00
G_M55611_IG07:  ;; offset=0x0030
            movprfx z0.b, p0/z, z0.b
                                                ;; size=4 bbWeight=1 PerfScore 2.00
G_M55611_IG08:  ;; offset=0x0034
            adr     x1, [G_M55611_IG12]
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG09:  ;; offset=0x0038
            add     x1, x1, x0,  LSL #3
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG10:  ;; offset=0x003C
            sub     x1, x1, #8
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG11:  ;; offset=0x0040
            br      x1
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG12:  ;; offset=0x0044
            asrd    z0.b, p0/m, z0.b, #1
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG13:  ;; offset=0x0048
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG14:  ;; offset=0x004C
            asrd    z0.b, p0/m, z0.b, #2
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG15:  ;; offset=0x0050
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG16:  ;; offset=0x0054
            asrd    z0.b, p0/m, z0.b, #3
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG17:  ;; offset=0x0058
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG18:  ;; offset=0x005C
            asrd    z0.b, p0/m, z0.b, #4
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG19:  ;; offset=0x0060
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG20:  ;; offset=0x0064
            asrd    z0.b, p0/m, z0.b, #5
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG21:  ;; offset=0x0068
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG22:  ;; offset=0x006C
            asrd    z0.b, p0/m, z0.b, #6
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG23:  ;; offset=0x0070
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG24:  ;; offset=0x0074
            asrd    z0.b, p0/m, z0.b, #7
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG25:  ;; offset=0x0078
            b       G_M55611_IG27
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG26:  ;; offset=0x007C
            asrd    z0.b, p0/m, z0.b, #8
                                                ;; size=4 bbWeight=1 PerfScore 3.00
G_M55611_IG27:  ;; offset=0x0080
            movz    xip0, #0x5678
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG28:  ;; offset=0x0084
            movk    xip0, #0x1234 LSL #16
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG29:  ;; offset=0x0088
            movk    xip0, #0xDEF0 LSL #32
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG30:  ;; offset=0x008C
            movk    xip0, #0x9ABC LSL #48
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG31:  ;; offset=0x0090
            ldr     xip1, [fp, #0x18]   // [V04 GsCookie]
                                                ;; size=4 bbWeight=1 PerfScore 2.00
G_M55611_IG32:  ;; offset=0x0094
            cmp     xip0, xip1
                                                ;; size=4 bbWeight=1 PerfScore 0.50
G_M55611_IG33:  ;; offset=0x0098
            beq     G_M55611_IG35
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG34:  ;; offset=0x009C
            bl      CORINFO_HELP_FAIL_FAST
                                                ;; size=4 bbWeight=1 PerfScore 1.00
G_M55611_IG35:  ;; offset=0x00A0
            ldp     fp, lr, [sp], #0x20
            ret     lr
                                                ;; size=8 bbWeight=1 PerfScore 2.00
G_M55611_IG36:  ;; offset=0x00A8
            bl      CORINFO_HELP_THROW_ARGUMENTOUTOFRANGEEXCEPTION
                                                ;; size=4 bbWeight=0 PerfScore 0.00
G_M55611_IG37:  ;; offset=0x00AC
            brk_windows #0
                                                ;; size=4 bbWeight=0 PerfScore 0.00
```